src/ExplainProvenanceImpl.h

```diff
@@ -619,29 +619,29 @@ class ExplainProvenanceImpl : public ExplainProvenance {
         // parse arguments in each relation Tuple
         for (size_t i = 0; i < rels.size(); ++i) {
             Relation* relation = prog.getRelation(rels[i].first);
-            // constTulpe stores value of constant argument in the relation tuple
+            // constTulpe stores number or index of symbol for constant arguments in tuple
             std::vector<RamDomain> constTuple;
-            // query relation does not exist
+            // relation does not exist
             if (relation == nullptr) {
-                queryResult += "Relation <" + rels[i].first + "> does not exist\n";
-                return queryResult;
+                std::cout << "Relation <" << rels[i].first << "> does not exist" << std::endl;
+                return;
             }
-            // airty error
+            // arity error
             if (relation->getArity() - 2 != rels[i].second.size()) {
-                queryResult += "<" + rels[i].first + "> has arity of " +
-                               std::to_string(relation->getArity() - 2) + "\n";
-                return queryResult;
+                std::cout << "<" + rels[i].first << "> has arity of "
+                          << std::to_string(relation->getArity() - 2) << std::endl;
```

Comment:
Could you change this (and also line 630) to use the relation's numberOfHeights instead of - 2? Like how it's done in line 449